src/components/views/settings/Notifications.js

```diff
@@ -37,240 +37,186 @@ var PushRuleVectorState = {
     OFF: "off"
 };
 
+// Encodes a dictionary of {
+//   "notify": true/false,
+//   "sound": string or undefined,
+//   "highlight: true/false,
+// }
+// to a list of push actions.
+function encodeActions(action) {
+    var notify = action.notify;
+    var sound = action.sound;
+    var highlight = action.highlight;
+    if (notify) {
+        var actions = ["notify"];
+        if (sound) {
+            actions.push({"set_tweak": "sound", "value": sound});
+        }
+        if (highlight) {
+            actions.push({"set_tweak": "highlight"});
+        } else {
+            actions.push({"set_tweak": "highlight", "value": false});
+        }
+        return actions;
+    } else {
+        return ["dont_notify"];
+    }
+}
+
+// Decode a list of actions to a dictionary of {
+//   "notify": true/false,
+//   "sound": string or undefined,
+//   "highlight: true/false,
+// }
+// If the actions couldn't be decoded then returns null.
+function decodeActions(actions) {
```

Comment:
Hmm, I'm not sure if they are exactly the same function since this gives up if it doesn't understand the tweaks, or actions.

But I agree that we should think about moving some of this code into the SDK so that other client authors don't need to duplicate it.